containers/src/Kokkos_DualView.hpp

```diff
@@ -182,8 +182,20 @@ class DualView : public ViewTraits< DataType , Arg1Type , Arg2Type, Arg3Type >
   //! \name Counters to keep track of changes ("modified" flags)
   //@{
 
-  View<unsigned int,LayoutLeft,typename t_host::execution_space> modified_device;
-  View<unsigned int,LayoutLeft,typename t_host::execution_space> modified_host;
+#ifndef KOKKOS_ENABLE_DEPRECATED_CODE
+private:
+  // modified_flags[0] -> host
+  // modified_flags[1] -> device
+  typedef View<unsigned int[2],LayoutLeft,Kokkos::HostSpace> t_modified_flags;
```

Comment:
Could `HostSpace::execution_space` ever differ from `t_host::execution_space::memory_space`?

Reply:
That question doesn't make sense since it will always differ. If you mean whether `t_host::execution_space` could be different from `HostSpace::execution_space` than yes. For a DualView in CudaUVMSpace `t_host::execution_space` is `Cuda`. If you mean whether `t_host::execution_space::memory_space` could be different from `HostSpace` same deal. Actually it is even worse. `std::is_same<MemorySpace::execution_space::memory_space,MemorySpace>::value` is not always true. For example for CudaUVMSpace the execution space is Cuda, which without having the `force_uvm` option `on` has `CudaSpace` as its memory space. Similar things are true for HostSpace, OpenMP, HBWSpace combinations.